jmxtrans-output/jmxtrans-output-log4j/pom.xml

```diff
@@ -47,10 +47,12 @@
 		<dependency>
 			<groupId>ch.qos.logback</groupId>
 			<artifactId>logback-classic</artifactId>
+			<optional>true</optional>
 		</dependency>
 		<dependency>
 			<groupId>ch.qos.logback</groupId>
 			<artifactId>logback-core</artifactId>
+			<optional>true</optional>
```

Comment:
The optional flag means that we compile against it, but it is not picked up as a dependency by modules which depend on this pom.  The logback components are linked to as if they were slf4j:slf4j-api, which has the exact same interface, but different semantics from the viewpoint of SLF4J.

If you find this confusing, you are not alone.  It took me months the first time I had to untangle an SLF4J build.